tests_e2e/tests/recover_network_interface/recover_network_interface.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# This test uses CSE to bring the network down and call check_and_recover_nic_state to bring the network back into an
+# 'up' and 'connected' state. The intention of the test is to alert us if there is some change in newer distros which
```

Comment:
how do we know change in new distros as we are validating on fixed distros.

Reply:
Eventually we want to attempt to recover the network interface for all distros where we restart the network interface, but this PR only implements the recovery for RedhatOSUtil (all fedora distros > 7 and < 8). 

There is a TODO in the yml for this scenario which says we should expand coverage to other distros once implemented, and then this comment will be accurate :)